galaxy_ng/app/models/contentguard.py

```diff
@@ -0,0 +1,25 @@
+from rest_framework.exceptions import APIException
+from pulpcore.plugin.models import ContentGuard
+
+
+class CollectionDownloadContentGuard(ContentGuard):
+    """
+    A content guard that protects content based on user authentication
+    """
+
+    def permit(self, request):
+        """
+        Authorize the specified request based on if the request is authenticated.
+        """
+        from galaxy_ng.app.api.v3.viewsets import CollectionArtifactDownloadView
+
+        view = CollectionArtifactDownloadView()
+        setattr(view, "get_object", lambda: self)
```

Comment:
This is overriding the `get_object` function so that it always returns the content guard instance. What we need to do is make it so that `get_object` returns the collection that is being accessed.

However, looking through the access policy, it looks like the download action doesn't take any conditions, so we can probably get away without defining a `get_object` method on the viewset until we get to a point where we need to add conditions to the download function.
```
        {
            "action": ["download"],
            "principal": 'authenticated',
            "effect": "allow",
        },
```